staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1439,6 +1441,13 @@ func (c *Cacher) Wait(ctx context.Context) error {
 	return c.ready.wait(ctx)
 }
 
+// setBookmarkNewFuncIfRequested sets bookmarkNewFunc field in watchCacheInterval for watchlist request
+func (c *Cacher) setBookmarkNewFuncIfRequested(cacheInterval *watchCacheInterval, opts storage.ListOptions) {
+	if opts.SendInitialEvents != nil && *opts.SendInitialEvents && opts.Predicate.AllowWatchBookmarks {
+		cacheInterval.bookmarkNewFunc = c.newFunc
```

Comment:
instead of setting `newFunc` here, let's create bookmark object directly here.

To do that, you need to also pass resourceVersion here, which will be just c.watchCache.resourceVersion

Reply:
Get it! I have changed it